pkgs/build-support/cc-wrapper/ld-wrapper.sh

```diff
@@ -155,7 +155,7 @@ fi
 
 # Only add --build-id if this is a final link. FIXME: should build gcc
 # with --enable-linker-build-id instead?
-if [ "$NIX_SET_BUILD_ID" = 1 ]; then
+if [ "$NIX_@infixSalt@_SET_BUILD_ID" = 1 ]; then
     for p in "${params[@]}"; do
         if [ "$p" = "-r" -o "$p" = "--relocatable" -o "$p" = "-i" ]; then
```

Comment:
Get rid of the repeated useless quoting. `[ "$p" = --relocatable ] ` also works (do not change the condition)